src/base.ts

```diff
@@ -29,6 +31,28 @@ export default abstract class extends Command {
         const {document} = await this.parser.parse(rawDocument);
         if (document !== undefined) {
           metadata = MetadataFromDocument(document, metadata);
+          const commandName : string = this.id || '';
+          switch (commandName) {
+          case 'validate':
```

Comment:
I'm a bit confused about the reason you moved all the logic from each command to this big switch statement here.

As I see this is going into a bit of deep hole, do you think we could rollback this and just merge the PR so we can move forward and start collecting metrics? Then we could work on this later.

The extended side of this comment, for future work:

We have what we would like to have **ideally**, a way of tracking, from the base.ts, that the command was executed and include some metadata, if exist. 
But I would say nothing about moving all the metadata collection logic to the base.ts file. This should stay in each command as it is responsibility of each command to collect such metadata. 

Instead of doing this, have you think about the possibility of setting all this metadata into a property declared in the base.ts file but set on each command? I mean, something like `this.metadata['success'] = true;` but from the command.